server.js

```diff
@@ -19,5 +20,48 @@ const messages = [welcomeMessage];
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
+app.post("/messages", (req, res) => {
+  //validate our input
+  if (!req.body.from || !req.body.text) {
+    res.status(400).send("from must be valid");
+    return;
+  }
+
+  // build our new resourse
+  const newMessage = req.body;
+  newMessage.id = uuidv4();
+  messages.push(newMessage);
+  res.status(201).json(newMessage);
+});
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.get("/messages/:id", (req, res) => {
+  console.log(req.params);
+  const id = req.params.id;
+  const msg = messages.find((message) => message.id === id);
+  if (msg === undefined) {
+    res.send("message not found");
```

Comment:
You should return this message with a 404 status, as many clients will only check status codes to do error handling. This will be a `200 message not found` response, which is weird.